web.py

```diff
@@ -97,14 +110,27 @@ def format(version):
 
 @route("/compile.json", method=["POST", "OPTIONS"])
 @enable_post_cors
+@extractor("backtrace", "0", ("0", "1", "2"))
 @extractor("syntax", "att", ("att", "intel"))
 @extractor("color", False, (True, False))
 @extractor("version", "stable", ("stable", "beta", "nightly"))
 @extractor("optimize", "2", ("0", "1", "2", "3"))
 @extractor("emit", "asm", ("asm", "llvm-ir", "mir"))
-def compile(emit, optimize, version, color, syntax):
+def compile(emit, optimize, version, color, syntax, backtrace):
     args = ["-C", "opt-level=" + optimize]
-    if optimize == "0":
+    if "0" == optimize:
+        debug = True
+    else:
+        debug = False
+    if "2" == backtrace:
+        if debug:
+            backtrace="1"
+        else:
+            backtrace="0"
+    if "1" == backtrace:
+        args.insert(0, "--backtrace")
+        #XXX: if exists, --backtrace must be the first arg passed to compile.sh
+    if debug:
```

Comment:
I wonder, can this and the above block share code for building up the arguments to the compiler?